tests/test_generator/test_lengths.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+
+from gutenTAG import GutenTAG
+
+
+class TestLengths(unittest.TestCase):
+    def setUp(self) -> None:
+        self.config = {
+            "timeseries": [
+                {
+                    "name": "test-amp",
+                    "length": 500,
+                    "base-oscillations": [{"kind": "sine"}],
+                    "anomalies": [
+                        {
+                            "position": "end",
+                            "length": 47,
+                            # "creeping-length": 12,  # remove to test the default 20% option
+                            "channel": 0,
+                            "kinds": [{"kind": "amplitude", "amplitude_factor": 2.0}],
+                        }
+                    ],
+                }
+            ]
+        }
+        self.creeping_factor = 0.3
+
+    def update_config(self, length: int, include_creeping: bool = False):
+        self.config["timeseries"][0]["anomalies"][0]["length"] = length
+        cl = int(round(length * self.creeping_factor)) if include_creeping else 0
+        self.config["timeseries"][0]["anomalies"][0]["creeping-length"] = cl
```

Comment:
I'm in favor of testing many lengths and abstracting this away as you do. However, I would avoid mutating members of the TestCase. I am unsure how this behaves when the tests are executed in parallel. To be on the safe side, please create a copy of `self.config`, mutate the copy, and just return the copy. You can then pass it to the `__run` function (see below).

Reply:
Thanks for the opportunity to learn about unittest with your comments 👍